lib/logstash/plugin_mixins/jdbc/statement_handler.rb

```diff
@@ -148,41 +148,28 @@ def perform_query(db, sql_last_value)
     private
 
     def build_query(db, sql_last_value)
-      @parameters = create_bind_values_hash
-      if statement_prepared.false?
-        prepended = parameters.keys.map{|v| v.to_s.prepend("$").to_sym}
-        @prepared = db[statement, *prepended].prepare(:select, name)
```

Comment:
review note: this `@prepared` ivar holds a reference to the `db` that was used to create it, which made it a source of connection leaks.